src/main/java/edu/cornell/kfs/module/cam/rest/resource/CuCapAssetInventoryApiResource.java

```diff
@@ -112,8 +115,48 @@ public Response getAsset(@PathParam(CuCamsConstants.CapAssetApi.ASSET_TAG_PARAME
         }
     }
 
+    @PUT
+    @Path("asset/inventory/{capital_asset_number}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response updateAsset(@PathParam(CuCamsConstants.CapAssetApi.CAPITAL_ASSET_NUMBER) String capitalAssetNumber, @Context HttpHeaders headers) {
+        try {
+            if (!validateUpdateAssetQueryParameters()) {
+                LOG.error("updateAsset: Bad Request for Asset Inventory #" + capitalAssetNumber + " " + servletRequest.getQueryString());
+                return respondBadRequest();
+            }
+
+            LOG.info("updateAsset: Requesting Capital Asset Inventory #" + capitalAssetNumber + " " + servletRequest.getQueryString());
+            String conditionCode = servletRequest.getParameter(CuCamsConstants.CapAssetApi.CONDITION_CODE);
+            String buildingCode = servletRequest.getParameter(CuCamsConstants.CapAssetApi.BUILDING_CODE);
+            String roomNumber = servletRequest.getParameter(CuCamsConstants.CapAssetApi.ROOM_NUMBER);
+            Asset asset = getCuAssetService().updateAssetInventory(capitalAssetNumber, conditionCode, buildingCode, roomNumber);
+            if (asset == null) {
+                LOG.error("updateAsset: Asset Inventory #" + capitalAssetNumber + " Not Found");
+                return respondNotFound();
+            }
+            LOG.info("updateAsset: Updated Capital Asset Inventory #" + capitalAssetNumber + " " + asset.getLastInventoryDate().toString());
+
+            Properties properties = getAssetProperties(asset);
+            return Response.ok(gson.toJson(properties)).build();
+        } catch (Exception ex) {
+            LOG.error("updateAsset", ex);
+            return ex instanceof BadRequestException ? respondBadRequest() : respondInternalServerError(ex);
+        }
+    }
+
+    private boolean validateUpdateAssetQueryParameters() {
+        List<String> paramNames = Collections.list(servletRequest.getParameterNames());
+        return paramNames.contains(CuCamsConstants.CapAssetApi.CONDITION_CODE) &&
+                paramNames.contains(CuCamsConstants.CapAssetApi.BUILDING_CODE) &&
+                paramNames.contains(CuCamsConstants.CapAssetApi.ROOM_NUMBER);
+    }
+
     private Properties getAssetProperties(Asset asset) {
         Properties assetProperties = new Properties();
+        assetProperties.put(CuCamsConstants.CapAssetApi.CAPITAL_ASSET_NUMBER, asset.getCapitalAssetNumber());
```

Comment:
Should this line use the "safelyAddProperty" method, like in the other lines below it?

Reply:
I don't think it needs the capital asset number because it is of type long.

Reply:
Is another variant of "safelyAddProperty" needed to check for non-null versions of non-string objects? The capital asset number is a Long wrapper object and not a primitive long.